pandas/tests/indexing/test_coercion.py

```diff
@@ -821,6 +821,9 @@ def test_replace_series(self, how, to_key, from_key):
         if (from_key.startswith('datetime') and to_key.startswith('datetime')):
             pytest.xfail("different tz, currently mask_missing "
                          "raises SystemError")
+        elif from_key in ['datetime64[ns, US/Eastern]', 'datetime64[ns, UTC]']:
+            pytest.xfail(reason='GH #18376, tzawareness-compat bug '
+                                'in BlockManager.replace_list')
```

Comment:
might be discussed before, but since you need to add a xfail here, is this introducing a regression?

Reply:
I guess?  See [above](https://github.com/pandas-dev/pandas/pull/18376#issuecomment-346969384):

> so replace_list is a giant hack ATM and needs to be fixed. maybe this is the impetus. However, don't really want to add your hack on top. What exactly is failing w/o you touching replace? let's isolate and x-fail those tests for now.

Reply:
Can you give an explicit code example that works now and will fail after this PR?

Reply:
@jorisvandenbossche good catch.  Two things here:

1) Does the fact that this is not immediately obvious by inspection suggest that test parametrization may have been taken a step too far?

2) In putting together an answer to this question I found that any non-datetime comparison raises, whereas I expect we want `DatetimeIndex(...) == "baz"` to just be Falsey (following the behavior of `Timestamp.__richcmp__` and `Timedelta.__richcmp__`).  So I need to fix this, will answer the original question after that.

Reply:
@jorisvandenbossche An example of a case that xfails after this PR:

```
index = pd.Index([3, 4], dtype='int64', name=u'xxx')
obj = pd.Series([pd.Timestamp('2011-01-01', tz='US/Eastern'),
                 pd.Timestamp('2011-01-03', tz='US/Eastern')],
                index=index, name='yyy')
replacer = pd.Series([pd.Timedelta(days=1), pd.Timedelta(days=2)],
                     index=obj)

>>> result = obj.replace(replacer)
[...]
TypeError: Cannot compare tz-naive and tz-aware datetime-like objects
```